### ISSUE TITLE
[Bug]: "Freetube cannot be closed. Please close it manually and click Retry to continue"

### ISSUE THREAD

**Issue by MegaBidoof** [FreeTubeApp/FreeTube#3324]
### Guidelines

- [ ] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

When running `freetube-0.18.0-setup-x64.exe` on Windows 10 to update from version 0.17.0 (currently installed) to v0.18.0, I get the following error message and the update fails. This has been happening ever since v0.18.0 was first released several months ago.
![image](https://user-images.githubusercontent.com/119820017/226121030-7a696dfd-9bbe-4d91-ac28-c93a3e7a9679.png)
When clicking "Retry", it loops back to the same error message.
When clicking "Cancel", the following pops up:
![image](https://user-images.githubusercontent.com/119820017/226121230-8cea422c-8a4e-4d84-911f-ee94ffa3e4b4.png)

When trying to run the installer for the latest nightly build (`freetube Setup 0.18.0-nightly-2759.exe`), I get the exact same error.

### Expected Behavior

v0.17.0 would be uninstalled, and v0.18.0 would be installed, while importing my current v0.17.0 settings.

### Issue Labels

accessibility issue

### FreeTube Version

v0.17.0

### Operating System Version

Windows 10

### Installation Method

.exe

### Primary API used

Invidious API

### Additional Information

Running as administrator makes no difference.
I have not tried uninstalling 0.17.0 manually, because I didn't want to lose my settings, nor risk being unable to use any version of FreeTube altogether (currently 0.17.0 works fine). But judging by the second error message, I'd say uninstalling it manually won't be possible either.

FreeTube is not opened. If it is, the error message is different, and allows me to close it directly with the click of a button. When this happens, the previous error still pops up, after closing the running window.

### Nightly Build

- [x] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @MegaBidoof did u try restarting your machine?

**Comment by MegaBidoof** [comment]
Just tried it again just to be sure. Run installer > Restart > Run installer. But the same error popped up both times the installer was run.
It was to be expected, though, as I've been facing this problem for several months now, with several sporadic attempts at installing.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
hi @MegaBidoof v0.19.0 has been released for a while now. Do you still experience this issue?

**Comment by MegaBidoof** [comment]
Firstly apologies for the delay in replying.

Secondly, yes, this issue is still happening. I have tried with both the 0.19.0 beta release installer and the latest nightly build installer (Build #3449). I am still getting exactly the same behavior.

I have searched for that error message and got some results regarding other different programs with the same error. 
Pardon me if what I am about to say is nonsense, but could it be related to Electron? I am not really familiar with it, but afaik FreeTube uses it, and I found [this possibly related thread](https://github.com/electron-userland/electron-builder/issues/6865).

In any case, thanks for your help.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@MegaBidoof not sure if this is Electron related, u could try out the latest nightly build. We recently updated to the latest version of Electron

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Lets keep this open until its gone but yes next step would be a uninstalling and installing latest stable version

Don't forget to backup your data
https://docs.freetubeapp.io/usage/data-location/#notable-files

**Comment by flying19880517** [comment]
https://github.com/electron-userland/electron-builder/issues/6865

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@MegaBidoof it has been a while. Are you still running into this?

**Comment by MegaBidoof** [comment]
I tried to uninstall and re-install but got the same error when trying to uninstall. I tried uninstalling via settings, control panel, command line, etc, to no avail. As the error suggested some conflict between the uninstaller and a running FreeTube session, I tried booting into safe mode and uninstalling there, but got the same error. What eventually worked was renaming the .exe. After renaming and rebooting, the uninstaller advanced past the point of the error. So I decided to abort the uninstaller and check whether I could now update, and it worked. After that update went through, I've been able to update normally with no issues.
I still don't know what caused the error in the first place.

So, for those in the same position as me, try:
- Rename FreeTube.exe in the FreeTube installation folder
- Reboot your computer
- Update or uninstall FreeTube

Should I close the issue or leave it open, as it appears more people are still having the same problem?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
You can leave it open. Its an upstream issue but they havent found a reliable way to reproduce it yet

**Comment by Alatarius** [comment]
I am having this issue with the new release 0.23.8. Issue still not fixed and Freetube is completely unusable unless updated
